linera-execution/tests/test_execution.rs

```diff
@@ -1667,3 +1660,90 @@ async fn test_close_chain() {
     .unwrap();
     assert!(view.system.closed.get());
 }
+
+/// Tests if an application can't transfer the tokens in the chain's balance while executing
+/// messages.
+#[tokio::test]
+async fn test_message_handler_cant_spend_chain_balance() -> anyhow::Result<()> {
+    let amount = Amount::ONE;
+
+    let mut state = SystemExecutionState::default();
+    state.description = Some(ChainDescription::Root(0));
+    state.balance = amount;
+    let mut view = state.into_view().await;
+
+    let mut applications = register_mock_applications(&mut view, 1).await?;
+    let (application_id, application, _, _) = applications
+        .next()
+        .expect("Caller mock application should be registered");
```

Comment:
Alternatively a version of `register_mock_application` which returns the application - no need to `.next().expect()`.

Reply:
I agree, I opened issue #2851 to track this.